garden-backend-service/src/api/routes/entrypoints.py

```diff
@@ -98,3 +71,75 @@ async def delete_entrypoint(
     else:
         # no error if not found so DELETE is idempotent
         return {"detail": f"No entrypoint found with DOI {doi}."}
+
+
+@router.put("/{doi:path}", response_model=EntrypointMetadataResponse)
+async def create_or_replace_entrypoint(
+    doi: str,
+    entrypoint_data: EntrypointCreateRequest,
+    db: AsyncSession = Depends(get_db_session),
+    user: User = Depends(authed_user),
+):
+    existing_entrypoint: Entrypoint | None = await Entrypoint.get(db, doi=doi)
+
+    if existing_entrypoint is None:
+        return await _create_new_entrypoint(entrypoint_data, db, user)
+
+    assert_deletable_by_user(existing_entrypoint, user)
+    # re-assign ownership if specified
+    if entrypoint_data.owner_identity_id is not None:
+        new_owner: User | None = await User.get(
+            db, identity_id=entrypoint_data.owner_identity_id
+        )
+        existing_entrypoint.owner = new_owner or user
+
+    try:
+        # naive update with all other values from payload
+        for key, value in entrypoint_data.model_dump(
+            exclude="owner_identity_id"
+        ).items():
+            setattr(existing_entrypoint, key, value)
+
+        await db.commit()
+    except IntegrityError as e:
+        await db.rollback()
+        raise HTTPException(
+            status_code=status.HTTP_409_CONFLICT,
+            detail=f"Integrity error occurred: {str(e)}",
+        ) from e
+    return existing_entrypoint
+
+
+async def _create_new_entrypoint(
+    entrypoint: EntrypointCreateRequest,
+    db: AsyncSession,
+    user: User,
+) -> Entrypoint:
+    # default owner is authed_user unless owner_identity_id is explicitly provided
+    owner: User = user
+    if entrypoint.owner_identity_id is not None:
+        explicit_owner: User | None = await User.get(
+            db, identity_id=entrypoint.owner_identity_id
+        )
+        if explicit_owner is not None:
+            owner = explicit_owner
+        else:
+            logger.warning(
+                f"No user found with Globus identity ID {entrypoint.owner_identity_id}. "
+                f"Assigning default ownership to {user.identity_id} ({user.username}). "
+            )
```

Comment:
Is the no-owner logic something we could get rid of after the migration? (Not indicating that this is bad, just curious)

Reply:
yeah that's more or less what I'm thinking, seemed like a reasonably convenient middle ground until we need a more sophisticated ownership model than "only one user can touch this"